### PR TITLE
test/new-e2e/tests/apm: add e2e tests for auto-versioning

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?
These tests are introduced to ensure that image id and git commit id are attached to Trace and Stats payloads.

### Motivation
In #22828 we added support for auto-versioning in the agent by attaching image id and git commit id to trace and stats payloads.

### CODE REVIEW

File: test/new-e2e/tests/apm/tests.go [hunk 1]
```diff
@@ -68,6 +70,57 @@ func testTracesHaveContainerTag(t *testing.T, c *assert.CollectT, service string
...
+	t.Log("Got traces", traces)
+	for _, tr := range traces {
+		for _, tp := range tr.TracerPayloads {
+			t.Log("Tracer Payload Tags:", tp.Tags["_dd.tags.container"])
+			ctags, ok := getContainerTags(t, tp)
+			assert.True(t, ok)
+			imageID, ok := ctags["image_id"]
+			assert.True(t, ok)
+			t.Logf("Got image ID: %v", imageID)
+			assert.Equal(t, "sha256:76909dd636e2ac542b2ff22b134e55861807ec3b88117d3d8573da5adb5b22d0", imageID)
```
Comment (ahmed-mez) [inline test/new-e2e/tests/apm/tests.go:87]: The hash is expected to change when the `tracegen` image is rebuilt. We ca make assertions on the format (e.g the string starts with `sha256:` and the rest is a 256-bit hex).

Another alternative is to get the real image sha from docker directly, but in that case we would start testing the correctness of the tagger which is not in the scope of these tests.

File: test/new-e2e/tests/apm/tests.go [hunk 2]
```diff
@@ -68,6 +70,57 @@ func testTracesHaveContainerTag(t *testing.T, c *assert.CollectT, service string
...
+	t.Helper()
+	stats, err := intake.Client().GetAPMStats()
+	assert.NoError(c, err)
+	assert.NotEmpty(c, stats)
+	t.Log("Got apm stats:", spew.Sdump(stats))
+	for _, p := range stats {
+		for _, s := range p.StatsPayload.Stats {
+			t.Log("Client Payload:", spew.Sdump(s))
+			t.Logf("Got image ID: %v", s.GetImageTag())
+			assert.Equal(t, "sha256:76909dd636e2ac542b2ff22b134e55861807ec3b88117d3d8573da5adb5b22d0", s.GetImageTag())
```
Comment (ahmed-mez) [inline test/new-e2e/tests/apm/tests.go:102]: I'm a bit lost here - shouldn't https://github.com/DataDog/datadog-agent/pull/22828 add the image sha (as opposed to the image tag) ?

**Review by ahmed-mez — APPROVED** [review]

**Review by pducolin — APPROVED** [review]
Generally I would add `t.Helper` in helper functions

**Pull request merged**